sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -178,150 +185,85 @@ export abstract class V4PositionManager {
+     * - else, encode INCREASE_LIQUIDITY. If it is on a NATIVE pool, encode a SWEEP. Finally encode a SETTLE_PAIR
      */
-    const calldatas = ['0x', '0x']
-    let value = toHex(0)
+    invariant(JSBI.greaterThan(position.liquidity, ZERO), 'ZERO_LIQUIDITY')
 
-    return {
-      calldata: Multicall.encodeMulticall(calldatas),
-      value,
-    }
-  }
+    const calldatas: string[] = []
+    const planner = new V4PositionPlanner()
 
-  /**
-   * Produces the calldata for completely or partially exiting a position
-   * @param position The position to exit
-   * @param options Additional information necessary for generating the calldata
-   * @returns The call parameters
-   */
-  public static removeCallParameters(_position: Position, _options: RemoveLiquidityOptions): MethodParameters {
-    /**
-     * cases:
-     * - if liquidityPercentage is 100%, encode BURN_POSITION and then TAKE_PAIR
-     * - else, encode DECREASE_LIQUIDITY and then TAKE_PAIR
-     */
-    const calldatas = ['0x', '0x']
+    const isMintAction = isMint(options)
 
-    return {
-      calldata: Multicall.encodeMulticall(calldatas),
-      value: toHex(0),
+    // Encode initialize pool.
+    if (isMintAction && shouldCreatePool(options)) {
+      // No planner used here because initializePool is not supported as an Action
+      calldatas.push(
+        V4PositionManager.encodeInitializePool(position.pool.poolKey, options.sqrtPriceX96!, options.hookData)
+      )
     }
-  }
 
-  public static collectCallParameters(_options: CollectOptions): MethodParameters {
-    /**
-     * Collecting in V4 is just a DECREASE_LIQUIDITY with 0 liquidity and then a TAKE_PAIR
-     */
-    const calldatas = ['0x', '0x']
+    // adjust for slippage
+    const maximumAmounts = position.mintAmountsWithSlippage(options.slippageTolerance)
+    const amount0Max = toHex(maximumAmounts.amount0)
+    const amount1Max = toHex(maximumAmounts.amount1)
+
+    // mint
+    if (isMintAction) {
+      const recipient: string = validateAndParseAddress(options.recipient)
+      planner.addMint(
+        position.pool,
+        position.tickLower,
+        position.tickUpper,
+        position.liquidity,
+        amount0Max,
+        amount1Max,
+        recipient,
+        options.hookData
+      )
+    } else {
+      // increase
+      planner.addIncrease(options.tokenId, position.liquidity, amount0Max, amount1Max, options.hookData)
+    }
 
-    return {
-      calldata: Multicall.encodeMulticall(calldatas),
-      value: toHex(0),
+    // need to settle both currencies when minting / adding liquidity
+    planner.addSettlePair(position.pool.currency0, position.pool.currency1)
+
+    // Any sweeping must happen after the settling.
+    let value: string = toHex(0)
+    if (options.useNative) {
+      invariant(position.pool.currency0.isNative || position.pool.currency1.isNative, 'NO_NATIVE')
```

Comment:
Do we have a constants file to move `NO_NATIVE` into?

Reply:
moved!